ts/packages/ts/src/commands/ExportCommand.ts

```diff
@@ -1,22 +1,44 @@
 import _ from 'lodash';
 import { Command } from '../Command';
 import { type EvalCommandReturn } from './EvalCommand';
+import { NAME, ConfigError } from '../utils';
 
 export type ExportCommandReturn = {
   [key: string]: string;
 };
 
 export type ExportCommandParameters = {
   pipe: EvalCommandReturn;
+  keys?: (key: string) => string;
 };
 
 export class ExportCommand extends Command<ExportCommandReturn> {
   constructor(public parameters: ExportCommandParameters) {
     super(parameters);
   }
 
+  private mutateKeys(result: EvalCommandReturn): EvalCommandReturn {
+    const { keys } = this.parameters;
+
+    if (!keys) {
+      return result;
+    }
+
+    return _.mapKeys(result, (current, key) => {
+      try {
+        const mutatedKey = keys(key);
+        if (!NAME(mutatedKey)) {
+          throw new ConfigError('invalid config key', `${mutatedKey} must be a valid name`);
+        }
+        return mutatedKey;
+      } catch (error) {
+        return key;
```

Comment:
Why are we returning the original key and not throwing an exception? There is no requirement for that in the issue and in terms of UX, it makes things very unclear for the user if they supply a function that doesn't pass `NAME` validation.

Reply:
There is no requirement to raise exception as well.

Refactored

Reply:
Test refactored accordingly